### ISSUE TITLE
PHP7.x compatibility issues in version 1.4.3

### ISSUE THREAD

**Issue by rikuheinaaho-Pinja** [paytrail/paytrail-for-adobe-commerce#64]
**Describe the bug**

There is still non-backwards compatible php syntax in the module which causes `bin/magento setup:di:compile` to fail with the error below. I suspect it to be caused by class properties being declared with the syntax `private ClassName $foo`
```
syntax error, unexpected 'private' (T_PRIVATE), expecting variable (T_VARIABLE)#0 /var/www/html/vendor/composer/ClassLoader.php(443): Composer\Autoload\ClassLoader::Composer\Autoload\{closure}('/var/www/html/v...')
#1 [internal function]: Composer\Autoload\ClassLoader->loadClass('Paytrail\\Paymen...')
#2 [internal function]: spl_autoload_call('Paytrail\\Paymen...')
#3 /var/www/html/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(134): class_exists('Paytrail\\Paymen...')
```
**To Reproduce**

Run `bin/magento setup:di:compile` with module version 1.4.3 and PHP 7.x

**Comment by loueranta-paytrail** [comment]
Thanks for reporting, we will fix this.

**Comment by SvanteAvenla** [comment]
It's due to the use of constructor promotion syntax in the constructor in https://github.com/paytrail/paytrail-for-adobe-commerce/blob/master/Observer/RestoreCart.php.

**Comment by bartoszkaluzny-solteq** [comment]
Issue has been fixed with version 1.4.4
